subsystem/kubeapi_test.go

```diff
@@ -2962,6 +2979,120 @@ var _ = Describe("[kube-api]cluster installation", func() {
+					Value:     "p1",
+				},
+				{
+					Operation: "append",
+					Value:     `p2="this is an argument"`,
+				},
+			}
+			infraEnvSpec.KernelArguments = kargs
+			infraEnvKey := types.NamespacedName{
+				Namespace: Options.Namespace,
+				Name:      infraNsName.Name,
+			}
+			deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+			infraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, waitForReconcileTimeout)
+			Expect(kubeKernelArgs(infraEnv)).To(Equal(kargs))
+		})
+
+		setAndCheck := func(kargs []v1beta1.KernelArgument) {
+			infraEnvKey := types.NamespacedName{
+				Namespace: Options.Namespace,
+				Name:      infraNsName.Name,
+			}
+
+			Eventually(func() error {
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				infraEnv.Spec.KernelArguments = kargs
+				return kubeClient.Update(ctx, infraEnv)
+			}, "10s", "1s").ShouldNot(HaveOccurred())
+			Eventually(func(g Gomega) {
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				g.Expect(expectedSpecKargs(infraEnv.Spec.KernelArguments)).To(Equal(kargs))
+				dbInfraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, 1)
+				g.Expect(kubeKernelArgs(dbInfraEnv)).To(Equal(kargs))
+			}, "30s", "2s").Should(Succeed())
+		}
+
+		It("Update kernel arguments", func() {
+			By("Register infra env")
+			infraEnvSpec.ClusterRef = nil
+			deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+
+			infraEnvKey := types.NamespacedName{
+				Namespace: Options.Namespace,
+				Name:      infraNsName.Name,
+			}
+			dbInfraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, waitForReconcileTimeout)
+			Expect(dbInfraEnv.KernelArguments).To(BeNil())
+			infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+			Expect(infraEnv).ToNot(BeNil())
+
+			By("Set with kernel arguments")
+			kargs1 := []v1beta1.KernelArgument{
+				{
+					Operation: "append",
+					Value:     "p1",
+				},
+				{
+					Operation: "append",
+					Value:     `p2="this is an argument"`,
+				},
+			}
+			setAndCheck(kargs1)
+
+			By("Set with different kernel arguments")
+			kargs2 := []v1beta1.KernelArgument{
+				{
+					Operation: "append",
+					Value:     "p3",
+				},
+				{
+					Operation: "append",
+					Value:     `p4="this is another argument"`,
+				},
+			}
+			setAndCheck(kargs2)
+
+			By("Clear kernel arguments")
+			setAndCheck(make([]v1beta1.KernelArgument, 0))
+		})
+		DescribeTable("unsupported cases",
+			func(operation, value string) {
+				infraEnvSpec.ClusterRef = nil
+				infraEnvKey := types.NamespacedName{
+					Namespace: Options.Namespace,
+					Name:      infraNsName.Name,
+				}
+				deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+				infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+				infraEnv.Spec.KernelArguments = []v1beta1.KernelArgument{
+					{
+						Operation: operation,
+						Value:     value,
+					},
+				}
+				Expect(kubeClient.Update(ctx, infraEnv)).To(HaveOccurred())
```

Comment:
what are you checking here? 
do you have webhooks that are validating the parameters? 
unless there are webhooks the api should not return an error, reconcile will detect the error and reflect it in the sync condition

Reply:
The API defines enums and pattern, so it should be rejected by the API.
Please see https://github.com/openshift/assisted-service/blob/4efed6128e9e854c41a69b262b9374c5768fcf3b/api/v1beta1/infraenv_types.go#L108-L119

Reply:
Anyway, the API returns error in these cases.